src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -2750,7 +2752,13 @@ impl Coordinator {
         let cluster_id = {
             let catalog = self.catalog();
             let cluster = match explainee {
-                Explainee::Dataflow(_) => catalog.active_cluster(ctx.session())?,
+                Explainee::Dataflow(_) => {
+                    if target_cluster != TargetCluster::Active {
```

Comment:
How would this happen? Things being auto routed seem like not the user's fault. What if the explanation is on something on the introspection cluster?

Reply:
I think the only way this could happen is a bug b/c they can't refer to the dataflow object without being on that cluster. If the user is trying to explain a dataflow that lives on the introspection cluster and they're already on the introspection cluster TargetCluster should be Active.

i think this'll all be moot when Alexander does https://github.com/MaterializeInc/materialize/issues/18089 so I'll revert this diff